src/main/scala/karazin/scala/users/group/week1/homework/Homework.scala

```diff
@@ -40,26 +40,77 @@ object Homework :
 
   object `Boolean Operators` :
 
-    def not(b: Boolean): Boolean = ???
+    def not(b: Boolean): Boolean = b match {
+      case true => false
+      case _ => true
+    }
 
-    def and(left: Boolean, right: Boolean): Boolean = ???
+    def and(left: Boolean, right: Boolean): Boolean = (left, right) match {

```

Comment:
This is quite good solution but in the requirements it's said that and should works like `&&`. In Your case `right' always is evaluated. I'd like you to add an additional test for 
`and(false, throw new Exception())` -- if `and` works as `&&` then `Exception` won't be thrown. Hint: remind about "call by value strategy".

Reply:
I made functions and & or called by value because in preface it's said that the functions should be eager. But I can make them lazy as built-in && and ||.